### PR TITLE
Added new version of the app

### PR DESCRIPTION
**Pull request by jjasghar**
- updated to a more verbose demo site
- added test-kitchen support for myapp

Signed-off-by: JJ Asghar <jj@chef.io>

### CODE REVIEW

File: getting-started/cookbooks/webinar1-myapp/.kitchen.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+suites:
+  - name: default
+    run_list:
+      - recipe[webinar1-myapp::default]
+    attributes:
+      gce:
+        instance:
+          attributes:
+            startup-script-url: "https://storage.googleapis.com/jj-test-kitchen/bootstrap.sh"
```
Comment (nelsonjr) [inline getting-started/cookbooks/webinar1-myapp/.kitchen.yml]: https://raw.githubusercontent.com/nelsonjr/chef-webinars/master/getting-started/client/bootstrap.sh

File: getting-started/cookbooks/webinar1-myapp/.kitchen.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        cpuPlatform: "Intel Broadwell"
+        hostname: "machine-1.c.jj-test-kitchen.internal"
```
Comment (nelsonjr) [inline getting-started/cookbooks/webinar1-myapp/.kitchen.yml]: machine-1.c.chef-webinar1.internal

File: getting-started/cookbooks/webinar1-myapp/.kitchen.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        image: "projects/ubuntu-os-cloud/global/images/ubuntu-1604-xenial-v20171011"
+        networkInterfaces:
+          accessConfigs:
+            externalIp: "35.197.121.139"
```
Comment (nelsonjr) [inline getting-started/cookbooks/webinar1-myapp/.kitchen.yml]: Does this IP need to be real?

File: getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html>
+  <body>
+    <?php
```
Comment (nelsonjr) [inline getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb]: The <?php ?> block has to be in line 1, before DOCTYPE

File: getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    Here's some useful data we get from <a href="https://docs.chef.io/ohai.html">ohai</a>.
+
+    <ul>
+      <li>We got the startup script from: <%= node['gce']['instance']['attributes']['startup-script-url'] %></li>
+      <li>Our CPU arch is: <%= node['gce']['cpuPlatform'] %></li>
+      <li>My hostname on GCP is: <%= node['gce']['hostname'] %></li>
+      <li>Finally, my machine image: <%= node['gce']['image'] %></li>
+    </ul>
+
+    <?php
```
Comment (nelsonjr) [inline getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb]: Align <?php and ?> at column 1.

File: getting-started/cookbooks/webinar1-myapp/.kitchen.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    run_list:
+      - recipe[webinar1-myapp::default]
+    attributes:
+      gce:
+        instance:
+          attributes:
+            startup-script-url: "https://storage.googleapis.com/jj-test-kitchen/bootstrap.sh"
+        cpuPlatform: "Intel Broadwell"
+        hostname: "machine-1.c.jj-test-kitchen.internal"
+        image: "projects/ubuntu-os-cloud/global/images/ubuntu-1604-xenial-v20171011"
```
Comment (nelsonjr) [inline getting-started/cookbooks/webinar1-myapp/.kitchen.yml]: projects/centos-cloud/global/images/centos-7-v20171003

**Review by nelsonjr — CHANGES_REQUESTED** [review]

**Comment by nelsonjr** [comment]
Cool Kitchen update... how do I run this on my machine? What do I need?

File: getting-started/cookbooks/webinar1-myapp/.kitchen.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
+---
+driver:
+  name: vagrant
```
Comment (nelsonjr) [inline getting-started/cookbooks/webinar1-myapp/.kitchen.yml:3]: Can you add a DEVELOPER.md file on the cookbook explaining how to test using Test Kitchen? (this is cool!)

**Review by nelsonjr — CHANGES_REQUESTED** [review]

File: getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    header('Content-Type: text/html');
+    header('Cache-Control: must-revalidate');
+    header('Cache-Control: no-cache');
+    ?>
+
+    <center>
+      <img src="https://www.chef.io/wp-content/uploads/2017/10/partner-heros-google-10.png">
+    </center>
+
+    <p>Hello from <%= node['hostname'] %>, with the IP address of <%= node['hostname'] %> on a <%= node['platform'] %> virtual machine.</p>
```
Comment (nelsonjr) [inline getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb]: Please wrap all content to 80-columns.

**Review by nelsonjr — CHANGES_REQUESTED** [review]

File: getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+    <h2>metadata from GCP</h2>
+
+    Here's some useful data we get from <a href="https://docs.chef.io/ohai.html">ohai</a>.
```
Comment (nelsonjr) [inline getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb]: Replace . with : at the end.

File: getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    header('Cache-Control: no-cache');
+    ?>
+
+    <center>
+      <img src="https://www.chef.io/wp-content/uploads/2017/10/partner-heros-google-10.png">
+    </center>
+
+    <p>Hello from <%= node['hostname'] %>, with the IP address of <%= node['hostname'] %> on a <%= node['platform'] %> virtual machine.</p>
+
+    <h2>metadata from GCP</h2>
```
Comment (nelsonjr) [inline getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb]: "M"

File: getting-started/cookbooks/webinar1-myapp/.kitchen.yml (hunk 1 above)
Comment (jjasghar) [reply getting-started/cookbooks/webinar1-myapp/.kitchen.yml]: done

File: getting-started/cookbooks/webinar1-myapp/.kitchen.yml (hunk 2 above)
Comment (jjasghar) [reply getting-started/cookbooks/webinar1-myapp/.kitchen.yml]: done

File: getting-started/cookbooks/webinar1-myapp/.kitchen.yml (hunk 6 above)
Comment (jjasghar) [reply getting-started/cookbooks/webinar1-myapp/.kitchen.yml]: done

File: getting-started/cookbooks/webinar1-myapp/.kitchen.yml (hunk 3 above)
Comment (jjasghar) [reply getting-started/cookbooks/webinar1-myapp/.kitchen.yml]: removed.

File: getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb (hunk 4 above)
Comment (jjasghar) [reply getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb]: fixed

File: getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb (hunk 5 above)
Comment (jjasghar) [reply getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb]: fixed

**Comment by nelsonjr** [comment]
@jjasghar looking good but there are some errors in the display. applying the cookbook (a) not all variables are there and (b) some values are not matching their claims:

```
Hello from webinar-20171016, with the IP address of webinar-20171016 on a centos virtual machine.

metadata from GCP

Here's some useful data we get from ohai.
We got the startup script from: gs://chef-webinar1/bootstrap.sh
Our CPU arch is:
My hostname on GCP is:
Finally, my machine image:
It's Mon, 16 Oct 2017 03:33:22 +0000 around here
```

**Review by nelsonjr — CHANGES_REQUESTED** [review]

File: getting-started/cookbooks/webinar1-myapp/.kitchen.yml (hunk 7 above)
Comment (jjasghar) [reply getting-started/cookbooks/webinar1-myapp/.kitchen.yml:3]: Done.

File: getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb (hunk 10 above)
Comment (jjasghar) [reply getting-started/cookbooks/webinar1-myapp/templates/default/index.php.erb]: fixed

**Pull request merged**